stagemonitor-core/src/main/java/org/stagemonitor/core/elasticsearch/ElasticsearchClient.java

```diff
@@ -502,12 +535,28 @@ public Void handleResponse(HttpRequest<?> httpRequest, InputStream inputStream,
 							boolean isNowAvailable = statusValue.equals("green") || statusValue.equals("yellow");
 							if (isNowAvailable) {
 								if (!isElasticsearchAvailable()) {
+									httpClient.send(HttpRequestBuilder.<Void>forUrl(elasticsearchUrl.toString())
+											.method("GET")
+											.successHandler(new HttpClient.ResponseHandler<Void>() {
+												@Override
+												public Void handleResponse(HttpRequest<?> httpRequest, InputStream inputStream, Integer statusCode, IOException e) throws IOException {
+													JsonNode clusterResponse = JsonUtils.getMapper().readTree(inputStream);
+													esVersion = clusterResponse.has("version") ? clusterResponse.get("version").get("number").asText() : "0.0.0";
+													logger.info(String.format("Elasticsearch Version: %s", esVersion));
+													return null;
+												}
+											})
+											.build());
 									logger.info("Elasticsearch is available again.");
 								}
+								elasticsearchAvailable.set(true);
```

Comment:
This line has to be after the invocation of all observers. Only then we can be sure that the index pattern is properly initialized and ready to receive spans and metrics.

Reply:
CorePlugin.initializePlugin is called during the Observer Initialization and checks the Elasticsearch availability. It will report FAIL if the flag is not set. I'll try to find a proper way to do this.